include/boost/beast/core/detail/any_buffers.hpp

```diff
@@ -66,14 +66,52 @@ class any_buffers
         : b_(b)
         , a_(a)
         {
+            value_type* dest = reinterpret_cast<value_type*>(&buffer_store_);
+            this->buffers = dest;
+            for(auto& buf : b_)
+            {
+                try
+                {
+                    new (dest) value_type (buf);
+                }
+                catch(...)
+                {
+                    unwind();
+                    throw;
+                }
+                ++this->size;
+            }
+        }
+
+        ~impl()
+        {
+            unwind();
         }
 
         Buffers b_;
         Alloc   a_;
 
+        // Dependency: ORDER
+        // buffers_ represents a variable-size array. It must be the last data member
+        // in this class.
+        typename std::aligned_storage<
+            sizeof(value_type),
+            alignof(value_type)>::type
+                buffer_store_[1];
```

Comment:
There's no need for this, you have to use `this+1` to get the storage past the end of the struct, and then work off that.

Reply:
wouldn't that potentially have alignment issues?

Reply:
you align the struct itself to the `value_type`. Boost.JSON has an example of it:
https://github.com/boostorg/json/blob/develop/include/boost/json/impl/array.hpp#L23